web/src/narrow.js

```diff
@@ -384,10 +384,28 @@ export function activate(raw_terms, opts) {
             // We override target_id in this case, since the user could be
             // having a near: narrow auto-reloaded.
             id_info.target_id = opts.then_select_id;
+            // Position selected row to not scroll off-screen.
             if (opts.then_select_offset === undefined) {
                 const $row = message_lists.current.get_row(opts.then_select_id);
                 if ($row.length > 0) {
-                    opts.then_select_offset = $row.get_offset_to_window().top;
+                    const row_props = $row.get_offset_to_window();
+                    const navbar_height = $("#navbar-fixed-container").height();
+                    // 30px height + 10px top margin.
+                    const compose_box_top = $("#compose").offset().top;
+                    const sticky_header_outer_height = 40;
+                    const min_height_for_message_top_visible =
+                        navbar_height + sticky_header_outer_height;
+
+                    if (
+                        row_props.top >= min_height_for_message_top_visible &&
+                        row_props.bottom <= compose_box_top
```

Comment:
This looks great for the case that we're off the top of the page, but for off-the-bottom where the compose box would occlude the message, I think we want to scroll the minimum amount such that the message is fully visible (but not so much that the top goes offscreen, in case the message is taller than the screen).

(Checked with @alya on this and she agrees). Can you do a revision to add that case as well, and post a screencast of it?

Reply:
https://github.com/zulip/zulip/assets/25124304/025cd269-8031-4080-bd62-ad76e5d43473

Fixed!